src/sentry/scim/endpoints/members.py

```diff
@@ -370,16 +373,29 @@ def post(self, request: Request, organization) -> Response:
         - Sentry's SCIM API doesn't currently support setting users to inactive,
         and the member will be deleted if active is set to `false`.
         - The API also does not support setting secondary emails.
+        - you can provide a sentryOrgRole to provision alongside the new user
         """
 
+        if (
+            features.has("organizations:scim-orgmember-roles", organization, actor=None)
+            and "sentryOrgRole" in request.data
+        ):
+            role = request.data.get("sentryOrgRole").lower()
+        else:
+            role = organization.default_role
+
+        allowed_roles = roles.get_all()
```

Comment:
This should be safe since the POST endpoint requires "member:admin" permissions.

Reply:
Can you elaborate on this? Are you saying that, since the endpoint itself requires the `"member:admin"` scope, anyone who is accessing it must have the highest clearance to set all roles?

If so, the Manager role also has `"member:admin"`. Just from eyeballing this, it looks as though this would would allow a Manager to promote to Owner, which is a bug.

Secondly, even if that's not a bug, I think it's good to avoid hard-coding assumptions that certain scopes do or don't belong to certain roles. Role definitions are config values that can change over time, plus `UserPermission`s could mess it up unexpectedly.

Reply:
So the way we prevent the self-promotion today is by doing a priority comparison between the org member's role and all the roles on the server; only allowing role assignment on roles with lower or equal priority.

The problem here is that the "user" assigning a member's role is an IDP integration and it doesn't have an org membership. So if we need to limit the roles that the integration can provision, we need to figure out a way to assign a role to the integration.

Reply:
Also, need to find a place to assign the role to the integration

Reply:
I see. Am I right in thinking that you only need to be a Manager to set up the integration? Is it a problem if a rogue Manager could set up a SCIM integration for the purpose of privilege-escalating a new account as an Owner? Or are we generally assuming that the pipeline of incoming SCIM data is trusted/controlled enough to give it unlimited discretion over all org memberships? (These might be product questions.)

Reply:
yeah, im curious how the manager -> owner privilege escalation can go now, as i see you can set Owner as the default role. wonder if we should make it so owner can't be a default role and close this gap.

Reply:
So we're gonna limit it to `member` and `billing` on SaaS. For here though, i'm gonna default to allowing every role.

Reply:
i could see admin being useful as well / not super dangeorus but I'll leave it up to the team to figure out!